presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -1363,6 +1375,18 @@ private ThriftMetastoreClient createMetastoreClient()
         return clientProvider.createMetastoreClient();
     }
 
+    private ThriftMetastoreClient createMetastoreClient(HiveContext context)
+            throws TException
+    {
+        ThriftMetastoreClient client = createMetastoreClient();
+        if (!isMetastoreImpersonationEnabled || hiveMetastoreAuthentication instanceof NoHiveMetastoreAuthentication) {
```

Comment:
It looks like `hiveMetastoreAuthentication` is only used here for this `instanceof` check, so we could move that into the constructor as part of the `isMetastoreImpersonationEnabled` field.